core/src/main/scala/io/delta/tables/DeltaTable.scala

```diff
@@ -807,7 +807,9 @@ object DeltaTable {
    */
   @Evolving
   def createOrReplace(spark: SparkSession): DeltaTableBuilder = {
-    new DeltaTableBuilder(spark, ReplaceTableOptions(orCreate = true))
+    DeltaTableUtils.withActiveSession(spark) {
+      new DeltaTableBuilder(spark, ReplaceTableOptions(orCreate = true))
```

Comment:
nit: put on the same line as the function definition
```suggestion
  def createOrReplace(spark: SparkSession): DeltaTableBuilder = DeltaTableUtils.withActiveSession(spark) {
      new DeltaTableBuilder(spark, ReplaceTableOptions(orCreate = true))
```
You can also import `DeltaTableUtils.withActiveSession` directly to shorten calls to `withActiveSession`, imo it's clear enough what `withActiveSession` does without fully qualifying it.